test/04-toggle-button-test.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+import ToggleButton from '../src/04/ToggleButton';
+
+const createComponent = (customProps = {}) => {
+  const props = Object.assign({}, {
+    onClick: () => {}
+  }, customProps);
+
+  return shallow( <ToggleButton {...props} />);
+}
+
+describe('Component: ToggleButton', () => {
+  describe('when it renders with default props', () => {
+    let component;
+    beforeEach(() => {
+      component = createComponent();
+    });
+
+    xit('contains two buttons', () => {
+      expect(component.find('button')).to.have.length(2);
+    });
+
+    xit('specifies which button is to the left', () => {
+      expect(component.find('button.toggle-button-left')).to.have.length(1);
+    });
+
+    xit('specifies which button is to the right', () => {
+      expect(component.find('button.toggle-button-right')).to.have.length(1);
+    });
+
+    xit('has a label', () => {
+      expect(component.find('label.toggle-label')).to.have.length(1);
+    });
+
+    xit('renders child elements with text content', () => {
+      component.children().nodes.forEach(node => expect(typeof node.props.children).to.equal('string'));
+    });
+
+    xit('toggles to false', () => {
+      expect(component.state().toggleOn).to.equal(false)
+    });
+
+    xit('changes toggleOn state onClick', () => {
+      component.simulate('click');
+      expect(component.state().toggleOn).to.equal(true)
+      component.simulate('click');
+      expect(component.state().toggleOn).to.equal(false)
```

Comment:
Line 48 and 49 might not be necessary. In the test above you assert that the default `toggleOn` state is false, and when you click the button it changes to true.

The test descriptions are easy to read if they are read as user actions. This one could perhaps be "toggles the button when clicking on it" or something like that.